voltdb/src/main/java/com/yahoo/ycsb/db/voltdb/VoltClient4.java

```diff
@@ -0,0 +1,315 @@
+ */
+public class VoltClient4 extends DB {
+
+  private static final String PROCEDURE_GET_WAS_NOT_FOUND = "Procedure Get was not found";
+  private Client mclient;
+  private byte[] mworkingData;
+  private ByteBuffer mwriteBuf;
+  private boolean useScanAll = false;
+
+  private static final Charset UTF8 = Charset.forName("UTF-8");
+
+  @Override
+  public void init() throws DBException {
+    Properties props = getProperties();
+    String servers = props.getProperty("voltdb.servers", "localhost");
+    String user = props.getProperty("voltdb.user", "");
+    String password = props.getProperty("voltdb.password", "");
+    String strLimit = props.getProperty("voltdb.ratelimit");
+    String useScanAllParam = props.getProperty("voltdb.scanall", "no");
+    
+    if (useScanAllParam.equalsIgnoreCase("YES")) { 
+      useScanAll = true;
+    }
+    
+    int ratelimit = strLimit != null ? Integer.parseInt(strLimit) : Integer.MAX_VALUE;
+    try {
+      mclient = ConnectionHelper.createConnection(Thread.currentThread().getId(), servers, user, password, ratelimit);
+
+      createYCSBDBIfNeeded();
+
+    } catch (Exception e) {
+      e.printStackTrace();
+      throw new DBException(e.getMessage());
+    }
+    mworkingData = new byte[1024 * 1024];
+    mwriteBuf = ByteBuffer.wrap(mworkingData);
+  }
+
+  /**
+   * Create the VoltDB YCSB DB if needed...
+   * 
+   * @throws Exception
+   */
+  private void createYCSBDBIfNeeded() throws Exception {
+
+    final String testString = "Test";
+
+    try {
+      ClientResponse response = mclient.callProcedure("Get", testString.getBytes(UTF8), testString);
+      if (response.getStatus() == ClientResponse.SUCCESS) {
+        // YCSB Database exists...
+        return;
+      } else {
+        throw new Exception(
+            "createYCSBDBIfNeeded(): got " + response.getStatusString() + " while testing for DB existence");
+      }
+    } catch (ProcCallException pce) {
+      if (pce.getMessage().equals(PROCEDURE_GET_WAS_NOT_FOUND)) {
+
+        YCSBSchemaBuilder ysb = YCSBSchemaBuilder.getInstance(mclient);
+
+        File tempDir = Files.createTempDirectory("voltdbYCSB").toFile();
+
+        ysb.loadClassesAndDDL(tempDir);
+
+      }
+
+    } catch (Exception e) {
+      e.printStackTrace();
+
+    }
+
+  }
+
+  /**
+   * @return true if we have a live DB connection
+   */
+  public boolean hasConnection() {
+
+    if (mclient != null && mclient.getConnectedHostList().size() > 0) {
+      return true;
+    }
+
+    return false;
+
+  }
+
+  @Override
+  public void cleanup() throws DBException {
+    ConnectionHelper.disconnect(Thread.currentThread().getId());
+  }
+
+  @Override
+  public Status delete(String keyspace, String key) {
+    try {
+      ClientResponse response = mclient.callProcedure("STORE.delete", key, keyspace.getBytes(UTF8));
+      return response.getStatus() == ClientResponse.SUCCESS ? Status.OK : Status.ERROR;
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
Please log via a logging framework instead of using printStackTrace directly

Reply:
Fixed in 0ebc1ec